### DISCUSSION TITLE
Powerful results from combining GraphQL and UQL

### DISCUSSION THREAD

**Discussion by rogerjbos** [grafana/grafana-infinity-datasource#268] [Show and tell]
Not sure I know the best way to write this up, but I have been struggling with producing a Grafana dashboard and the Infinity plugin really came to the rescue.  Using the GraphQL feature I was able to pull in the data I needed without resorting to loading it into a database, but I could not figure out how to do summarize or group by actions on it.  I saw that UQL had those functions, but I could not get it to work with my with my query.  @yesoreyeram answered my question and even created an example to show me how to do it, which allowed me to create this dashboard: 

<img width="1187" alt="Screen Shot 2022-02-22 at 11 40 49 PM" src="https://user-images.githubusercontent.com/4925977/155262828-26e82ea6-f652-4a02-9a86-80780cea5cf0.png">

Some of the nice features of UQL are that you can specify the data path once so you don't have to do it for every item:
```
parse-json
| scope "data.query.dexActions.nodes"
```
I was able to create an hour variable to aggregate by data over:
```
| extend "hour"=startofhour("timestamp")
```
Then I was able to format it to make it shorter and look better:
```
| extend "hour"=format_datetime("hour",'MM/DD hh:00')
```
Finally, I was able to summarize using a group by:
```
| summarize "Total volumeUSD / Hour"=sum("volumeUSD") by "hour"
```
This is really an amazing plugin.  For more details, please see the original question and answer with example.

**Comment by yesoreyeram** [comment]
Thank you @rogerjbos for the kind words and your time to write this.
